File: atcoder.jp/contests/dwacon6th-prelims/tasks/dwacon6th_prelims_a/main.py
from itertools import dropwhile
from operator import itemgetter
from typing import Tuple


def input_st() -> Tuple[str, int]:
    s, t = input().split()
    return s, int(t)


N = int(input())
ST = [input_st() for _ in range(N)]
X = input()

it = dropwhile(lambda st: st[0] != X, ST)
next(it)
ans = sum(map(itemgetter(1), it))

print(ans)

File: atcoder.jp/contests/dwacon6th-prelims/tasks/dwacon6th_prelims_a/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const sum = (array: number[]) => array.reduce((a, b) => a + b, 0);

const main = () => {
  const N = +input();
  const ST = [...Array(N)].map(() =>
    (([s, t]) => [s, +t] as [string, number])(input().split(" "))
  );
  const X = input();

  const i = ST.findIndex(([s]) => s === X);
  const ans = sum(ST.slice(i + 1).map(([, t]) => t));

  console.log(ans);
};

main();
